Framework/API/src/FileFinder.cpp

```diff
@@ -605,12 +605,32 @@ FileFinderImpl::findRuns(const std::string &hintstr,
       if (runEndNumber < runNumber) {
         throw std::invalid_argument("Malformed range of runs: " + *h);
       }
+      std::string previous_path{""};
```

Comment:
For strings the default constructor will give an empty string so we can collapse this to:
```c++
std::string previousPath, previousExt;
```
Note that I've switched to camelCase to be consistent with the coding standards.